libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -27,17 +28,19 @@ private const val RASTER_SOURCE_TYPE = "raster"
  *
  * Once instantiated, the controller will immediately start caching guidance data.
  *
- * In order to start caching map data, provide an instance via [setMapInstance].
- * At the moment, there can only be one instance of the map caching resources at a time.
+ * In order to start caching map data, provide an instance via [createMapControllers].
+ * To specify sources to cache, pass a list of id's via [createMapControllers].
+ * Source id's should look like "mapbox://mapbox.satellite", "mapbox://mapbox.mapbox-terrain-v2".
+ * The system only supports source hosted on Mapbox Services which URL starts with "mapbox://".
+ * If no ids are passed all available style sources will be cached.
+ *
  * The controller as well as [MapboxNavigation] instance it's holding can have
- * a different lifecycle than the [MapboxMap] instance, so make sure to call [removeMapInstance]
+ * a different lifecycle than the [MapboxMap] instance, so make sure to call [removeMapControllers]
  * whenever the [MapView] is destroyed to avoid leaking references or downloading unnecessary
- * resources. When the map instance is recreated, set it back with [setMapInstance].
+ * resources. When the map instance is recreated, set it back with [createMapControllers].
  *
  * The map instance has to be configured with a valid [TileStore] instance at the same path that was provided to [RoutingTilesOptions.filePath].
```

Comment:
This has to be updated. We want `RoutingTilesOptions.tileStore` to be the same rather than `RoutingTilesOptions.filePath`. Let's also mention that in the changelog as a breaking change.

Reply:
And what happens if they are different? Do we dubplate the data? Should we have a runtime check in `createMapControllers` that checks if the tile store instance is the same as the instance that Nav SDK has?

Reply:
> Do we duplicate the data or consumed resources?

I think no, maps and nav use different tiles, so maps will use tiles from the one store, nav from another one.

> Should we have a runtime check

not sure it's SDK responsibility, we have clear docs (I hope :)

Reply:
If you have separate instances, you will be supporting 2 different database instances. The storage place should be different too (different directories). Though it may not cause too many negative effects and significant resource duplication at the moment, we would prefer if it's the same instance so that the resources can be shared in 1 place. cc @kkaefer